src/main/java/uk/gov/ons/ctp/common/util/MultiIsoDateFormat.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.ons.ctp.common.util;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+
+public class MultiIsoDateFormat extends AggregatedDateFormat {
```

Comment:
We probably don't need AggregatedDateFormat and MultiIsoDateFormat. It's unlikely we will need to implement multiple versions of aggregated date formats and if we do can we not do it when we have that requirement?

Reply:
LOL when I copy & pasted MultiIsoDateFormat from rm-collection-exercise I forgot to paste the unit tests. Will do it now
My view re. subclassing is: if you are going to change a common library then make sure you do it properly.  If we need to change date formats etc then it would be nice to have the option to avoid re-releasing the common library

Reply:
As it stands you could just configure a spring bean to use different formats, which wouldn't be possible if you removed AggegatedDateFormat

Reply:
¯\_(ツ)_/¯ static factory method to do that in AggregatedDateFormat? It's not a bigger either way

Reply:
TBH that would be polishing a turd as the class that uses this (CustomObjectMapper) is referenced directly everywhere via new (no factory and no reference to the base ObjectMapper class). A factory method would still leave the client coupled to AggregatedDateFormat even if it's not using the constructor. You'd need an abstract factory class and that's giving me flashbacks to the bad old days :-P . I don't mind changing it but I'd prefer to leave it as is

Reply:
Unit test class added BTW